### PR TITLE
Add option to print module dependencies on data

### PR DESCRIPTION
**Pull request by wmtan**
This is a new feature requested by users in issue #12125  to add an option to cmsRun to print module dependencies in the configuration file to the output log. This PR also adds a unit test for the new feature, and adds some missing const qualifiers to prior code that is relevant to this feature.

### CODE REVIEW

**Comment by wmtan** [comment]
@Dr15Jones I will refrain from requesting testing of this feature until you have had a change to comment.

File: FWCore/Framework/interface/stream/EDAnalyzerAdaptorBase.h [hunk 1]
```diff
@@ -92,8 +92,10 @@ namespace edm {
       
       const EDConsumerBase* consumer() const;
       
-      void modulesDependentUpon(const std::string& iProcessName,
-                                std::vector<const char*>& oModuleLabels) const;
+      void modulesDependentUpon(std::string const& iProcessName,
+                                std::string const& iModuleLabel,
+                                bool print,
```
Comment (Dr15Jones) [inline FWCore/Framework/interface/stream/EDAnalyzerAdaptorBase.h]: it would be nice if this were ```iPrint``` to match the other arguments.

Comment (wmtan) [inline FWCore/Framework/interface/stream/EDAnalyzerAdaptorBase.h]: @Dr15Jones I changed all the "print" arguments to "iPrint".

**Comment by wmtan** [comment]
@cmsbuild please test

**Comment by wmtan** [comment]
@cmsbuild @Dr15Jones This build error occurs in the base IB and has nothing to do with this PR.

**Comment by Dr15Jones** [comment]
+1

**Comment by wmtan** [comment]
@cmsbuild please test

**Comment by wmtan** [comment]
@cmsbuild @Dr15Jones Whoops! The shell script file did not have execute permissions, which is why the test failed. I have fixed this now.

**Comment by wmtan** [comment]
@cmsbuild please test

**Comment by Dr15Jones** [comment]
The test failure is unrelated to thIs pull request.

**Comment by Dr15Jones** [comment]
+1

**Comment by fwyzard** [comment]
hi @wmtan , is there a way to send only the ModuleDependency messages to a log file ?

**Comment by wmtan** [comment]
@fwyzard Each such message contains the string "ModuleDependency" and is on a single line. So, you can use "grep".

**Comment by fwyzard** [comment]
Silly me, I though CMS used the MessageLogger to handle log messages, I hadn't though I would have to write my own.

**Comment by fwyzard** [comment]
By the way, could you change the heading from 
```
<ModuleDependency>
```
to
```
ModuleDependency
```
for consistency with the TimeReport and TrigReport messages ?

**Comment by wmtan** [comment]
@fwyzard The messages do use the MessageLogger, and are all in the category "ModuleDependency".
From reading the TWIKI from the message logger, I see no way to select  messages by category. I see only a way to limit them by category.

**Comment by wmtan** [comment]
@fwyzard No problem changing "<ModuleDependency>" to "ModuleDependency".

**Comment by wmtan** [comment]
@fwyzard PR #12418 changes the format according to your request.

**Comment by fwyzard** [comment]
Hi Bill,
thanks,
.A

**Pull request merged**